v3/src/components/axis/helper-models/date-axis-helper.ts

```diff
@@ -0,0 +1,447 @@
+import { Selection } from "d3"
+import { AxisHelper } from "./axis-helper"
+import { MutableRefObject } from "react"
+import { kAxisGap, kAxisTickLength, kDefaultFontHeight } from "../../graph/graphing-types"
+import { isDateAxisModel, IDateAxisModel } from "../models/axis-model"
+import {convertToDate, createDate, determineLevels, EDateTimeLevel, shortMonthNames}
+          from "../../../utilities/date-utils"
+import { measureTextExtent } from "../../../hooks/use-measure-text"
+
+type ILabelDateAndString = {
+  labelDate: Date
+  labelString: string
+}
+
+const getLevelLabelForValue = (level: EDateTimeLevel, date: Date | null): ILabelDateAndString => {
```

Comment:
We haven't generally used `null` values in CODAP, but I see I approved it in Piotr's date utilities code so let's not worry about it for the purpose of this PR.